blocks/graph.py

```diff
@@ -47,6 +48,7 @@ def recursion(current):
                 if application_call not in self.application_calls:
                     self.application_calls.add(application_call)
                     for av in application_call.auxiliary_variables:
+                        self.auxiliary_variables.add(av)
                         av.tag.application_call = current.tag.application_call
```

Comment:
@rizar @janchorowski Is there any reason this tagging happens here and not in the `add_auxiliary_variable` method? We could simply do `expression.tag.application_call = self` over there, no?